From stern@rowland.harvard.edu Tue May 30 07:12:34 2006
Date: Tue, 30 May 2006 10:12:30 -0400 (EDT)
From: Giridhar Pemmasani <giri@lmc.cs.sunysb.edu>
To: Greg KH <greg@kroah.com>, Giridhar Pemmasani <giri@lmc.cs.sunysb.edu>
Subject: usbcore: Fix broken RNDIS config selection
Message-ID: <Pine.LNX.4.44L0.0605301009410.5821-100000@iolanthe.rowland.org>

From: Giridhar Pemmasani <giri@lmc.cs.sunysb.edu>

RNDIS devices don't get configured owing to a typo in
choose_configuration().  This patch from Giridhar Pemmasani fixes the
typo.

From: Giridhar Pemmasani <giri@lmc.cs.sunysb.edu>
Signed-off-by: Alan Stern <stern@rowland.harvard.edu>
Signed-off-by: Greg Kroah-Hartman <gregkh@suse.de>

---
 drivers/usb/core/hub.c |    2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

--- gregkh-2.6.orig/drivers/usb/core/hub.c
+++ gregkh-2.6/drivers/usb/core/hub.c
@@ -1241,7 +1241,7 @@ static int choose_configuration(struct u
 				&& desc->bInterfaceClass == USB_CLASS_COMM
 				&& desc->bInterfaceSubClass == 2
 				&& desc->bInterfaceProtocol == 0xff) {
-#ifndef CONFIG_USB_NET_RNDIS
+#ifndef CONFIG_USB_NET_RNDIS_HOST
 			continue;
 #else
 			best = c;
